### PR TITLE
Add: Website buttons to AI/GS and Baseset settings windows.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

AI/GS have URLs which are displayed when selecting an AI/GS, but in the AI/GS settings windows there is no "Visit Website" button to simply open the URL.

Basesets do not have any URLs but could benefit from doing so.

The NewGRF settings and Content Download windows have "Visit Website" buttons the layout is inconsistent.

## Description

Add "Visit Website" button to AS and GS settings windows This also tweaks the layout of the other surrounding buttons.

Add support for "url" entry in BaseSet "[metadata]" section, and add a "Visit Website" button when available. No Basesets have this metadata yet and it's not clear if support is needed in bananas-api.

Lastly, tweak the layouts around the "Visit Website" buttons in the  Content Download and NewGRF settings for consistency.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/ai/ai_gui.cpp [hunk 1]
```diff
@@ -54,24 +54,28 @@ static const NWidgetPart _nested_ai_config_widgets[] = {
...
-					NWidget(WWT_PUSHTXTBTN, COLOUR_YELLOW, WID_AIC_CHANGE), SetFill(1, 1), SetMinimalSize(93, 0), SetDataTip(STR_AI_CONFIG_CHANGE_AI, STR_AI_CONFIG_CHANGE_TOOLTIP),
-					NWidget(WWT_PUSHTXTBTN, COLOUR_YELLOW, WID_AIC_CONFIGURE), SetFill(1, 1), SetMinimalSize(93, 0), SetDataTip(STR_AI_CONFIG_CONFIGURE, STR_AI_CONFIG_CONFIGURE_TOOLTIP),
-					NWidget(WWT_PUSHTXTBTN, COLOUR_YELLOW, WID_AIC_TEXTFILE + TFT_README), SetFill(1, 1), SetResize(1, 0), SetDataTip(STR_TEXTFILE_VIEW_README, STR_NULL),
+			NWidget(NWID_HORIZONTAL), SetPIP(0, WidgetDimensions::unscaled.hsep_wide, 0),
+				NWidget(NWID_VERTICAL, NC_EQUALSIZE),
+					NWidget(WWT_PUSHTXTBTN, COLOUR_YELLOW, WID_AIC_CHANGE), SetFill(1, 1), SetDataTip(STR_AI_CONFIG_CHANGE_AI, STR_AI_CONFIG_CHANGE_TOOLTIP),
+					NWidget(WWT_PUSHTXTBTN, COLOUR_YELLOW, WID_AIC_CONTENT_DOWNLOAD), SetFill(1, 1), SetDataTip(STR_INTRO_ONLINE_CONTENT, STR_INTRO_TOOLTIP_ONLINE_CONTENT),
 				EndContainer(),
-				NWidget(NWID_HORIZONTAL, NC_EQUALSIZE),
-					NWidget(WWT_PUSHTXTBTN, COLOUR_YELLOW, WID_AIC_CLOSE), SetFill(1, 1), SetMinimalSize(93, 0), SetDataTip(STR_AI_SETTINGS_CLOSE, STR_NULL),
```
Comment (rubidium42) [inline src/ai/ai_gui.cpp:70]: Shouldn't `WID_AIC_CLOSE` be removed from the enumeration as well? And everywhere where `WID_AIC_CLOSE` is used.

File: src/game/game_gui.cpp [hunk 2]
```diff
@@ -37,35 +37,38 @@ static const NWidgetPart _nested_gs_config_widgets[] = {
...
+						NWidget(WWT_PUSHTXTBTN, COLOUR_YELLOW, WID_GSC_TEXTFILE + TFT_LICENSE), SetFill(1, 1), SetResize(1, 0), SetDataTip(STR_TEXTFILE_VIEW_LICENCE, STR_NULL),
+					EndContainer(),
 				EndContainer(),
-				NWidget(WWT_PUSHTXTBTN, COLOUR_YELLOW, WID_GSC_CONTENT_DOWNLOAD), SetFill(1, 1), SetResize(1, 0), SetMinimalSize(279, 0), SetDataTip(STR_INTRO_ONLINE_CONTENT, STR_INTRO_TOOLTIP_ONLINE_CONTENT),
 			EndContainer(),
 		EndContainer(),
-	EndContainer(),
-	NWidget(NWID_HORIZONTAL),
-		NWidget(NWID_HORIZONTAL, NC_EQUALSIZE),
-			NWidget(WWT_PUSHTXTBTN, COLOUR_MAUVE, WID_GSC_ACCEPT), SetFill(1, 0), SetResize(1, 0), SetDataTip(STR_AI_SETTINGS_CLOSE, STR_NULL),
```
Comment (rubidium42) [inline src/game/game_gui.cpp:65]: Shouldn't `WID_GSC_ACCEPT` be removed from the enumeration as well? And everywhere where `WID_GSC_ACCEPT` is used.

**Comment by PeterN** [comment]
Missing screenshots... so Game Options:

image

NewGRF settings:

image

AI settings:

image

GS settings:

image

**Review by 2TallTyler — APPROVED** [review]
Big improvement.

Unused string checker isn't failing, so...where is `STR_AI_SETTINGS_CLOSE` still used? 🤔

**Pull request merged**